### PR TITLE
ci: separate parity/benchmarks

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Separate running benchmarks from unit tests as they turned out to be flaky/sensitive on the actual machine and eventually concurrently running jobs

lets also use upstreamed `RunIf` via #17504

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @carmocca @borda

### CODE REVIEW

File: tests/benchmarks_fabric/__init__.py
Comment (awaelchli) [inline tests/benchmarks_fabric/__init__.py:1]: These aren't benchmarks. We already have a folder called parity tests, let's keep it there please. 
Calling it benchmarks would be misleading.

Comment (lantiga) [reply tests/benchmarks_fabric/__init__.py:1]: I agree, let's go with parity as a general term @Borda

**Review by lantiga — COMMENTED** [review]
Let's change from `benchmarks` to `parity`

**Comment by Borda** [comment]
@lantiga lets get this merge and address the loss difference in another PR
as benchmarks are not required tests (they even were not) so it won't block any dev. 🐿️

**Pull request merged**